### PR TITLE
[ci] only compile necessary modules for self hosted pipeline

### PR DESCRIPTION
**Pull request by whhe**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -581,5 +608,13 @@ under the License.
                 </pluginManagement>
             </build>
         </profile>
+        <profile>
+            <id>self-hosted-pipeline</id>
+            <modules>
+                <module>flink-connector-debezium</module>
+                <module>flink-connector-oceanbase-cdc</module>
+                <module>flink-sql-connector-oceanbase-cdc</module>
```
Comment (ruanhang1993) [inline pom.xml:616]:
```
<module>flink-cdc-base</module>
<module>flink-connector-debezium</module>
<module>flink-connector-test-util</module>
<module>flink-connector-oceanbase-cdc</module>
<module>flink-sql-connector-oceanbase-cdc</module>
```

File: pom.xml [hunk 2]
```diff
@@ -477,8 +454,58 @@ under the License.
     </build>
 
     <profiles>
+        <profile>
+            <id>default</id>
```
[the remark this answers is not in the document]
Comment (whhe) [reply pom.xml:458]: > The `default` and `release` are totally same ? If true do we need `release` profile?

The only difference is that there are some plugins for releasing in 'release' profile. I'm not familar about the releasing of Flink CDC, if we remove the release profile here, these configurations can still be set in local settings.xml.

[the remark this answers is not in the document]
Comment (leonardBang) [reply pom.xml:458]: Got it

**Review by leonardBang — APPROVED** [review]
+1

**Pull request merged**